lib/launcher/src/Cardano/Launcher/Node.hs

```diff
@@ -132,22 +138,28 @@ withCardanoNode
     -> IO (Either ProcessHasExited a)
 withCardanoNode tr cfg action = do
     let socketPath = nodeSocketPath (nodeDir cfg)
-    cp <- cardanoNodeProcess cfg socketPath
-    withBackendCreateProcess tr cp $ \_ _ -> action $ CardanoNodeConn socketPath
+    let run output = do
+            cp <- cardanoNodeProcess cfg output socketPath
+            withBackendCreateProcess tr cp
+                $ \_ _ -> action $ CardanoNodeConn socketPath
+    case nodeOutputFile cfg of
+        Nothing -> run Inherit
+        Just file ->
+            withFile file AppendMode $ \h -> run (UseHandle h)
 
 {-------------------------------------------------------------------------------
                                     Helpers
 -------------------------------------------------------------------------------}
 
 -- | Generate command-line arguments for launching @cardano-node@.
-cardanoNodeProcess :: CardanoNodeConfig -> FilePath -> IO CreateProcess
-cardanoNodeProcess cfg socketPath = do
+cardanoNodeProcess :: CardanoNodeConfig -> StdStream -> FilePath -> IO CreateProcess
+cardanoNodeProcess cfg output socketPath = do
```

Comment:
Hrm. 🤔 There is some duplication here: the data in the argument `output` appears twice now:

* in the `nodeOutputFile` field of the `cfg :: CardanoNodeConfig` argument
* in the `output :: StdStream` argument

This is fine as long as the function `cardanoNodeProcess` is internal to this module, and callers are only allowed to use `withCardanoNode` to start a node. In other words — could you remove `cardanoNodeProcess` from the export list?